src/common/motor.rs

```diff
@@ -141,6 +171,18 @@ impl Motor for FakeMotor {
         self.power = pct;
         Ok(())
     }
+    fn go_for(&mut self, rpm: f64, revolutions: f64) -> anyhow::Result<()> {
+        // get_max_rpm
+        let (pwr, dur) = go_for_math(self.max_rpm, rpm, revolutions).unwrap(); 
+        if revolutions == 0.0 {
+            self.set_power(pwr)?;
+        } else {
+            self.set_power(pwr)?;
+            std::thread::sleep(dur); 
```

Comment:
we can't use thread sleep here because it will prevent the thread running the executor from making progress. We could return a Future to be polled to completion at an higher level